query.go

```diff
@@ -782,7 +784,33 @@ func (s *ChainService) GetCFilter(blockHash chainhash.Hash,
 					numFilters)
 			}
 
-			if s.persistToDisk {
+			persistChan <- resp
+		}
+	}()
+
+	if s.persistToDisk {
+		// Persisting to disk is the bottleneck for fetching filters.
+		// So we run the persisting logic in a separate goroutine so
+		// that we can unlock the mtxCFilter mutex as soon as we are
+		// done with caching the filters in order to allow more
+		// GetCFilter calls from the caller sooner.
+		go func() {
+			var (
+				resp *filterResponse
+				ok   bool
+			)
+
+			for {
+				select {
+				case resp, ok = <-persistChan:
```

Comment:
I think would be cleaner to send the filter to persist, instead of relying on a chan close above as a signal to check a local closure variable.